enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,295 @@
+ - next, Red Hats's docker daemon installed in 3.x had specific logic that auto-mounted the host's subscriptions 
+ into the container
+
+In V4.x, with 
+
+- the removal of the docker daemon
+- and the total replacement of the openshift ansible based installer
+
+all those 3.x mechanisms are gone, and a series of steps from either the cluster admin and/or developer attempting
+to run OpenShift Builds leveraging entitled content must be performed after install to enable access to the entitled
+content during the build.
+
+The highlights of those steps from [the current 4.x doc](https://docs.openshift.com/container-platform/4.3/builds/running-entitled-builds.html):
+
+1) First, either
+ 
+- create an ImageStreamTag to the UBI image from `registry.redhat.io` in the `openshift` namespace, or 
+- reference the UBI image directly and include a pull secret to `registry.redhat.io` in your BuildConfig
+
+2) Next, based on your access to the credentials:
+
+- With direct access to the entitlement pem files, create a generic secret from the files and reference the secret in
+your BuildConfig's input/source secret list
+- With Subscription Manager, and S2I build strategy, create config maps for the Subscription Manager configuration and 
+certs and reference the config maps in your BuildConfig's input/source config map list
+- With Subscription Manager, and Docker build strategy, you have to modify your Dockerfile and manually copy and 
+initialize as needed the information from the base image
+- With Satellite, and S2I build strategy, create a config map to reference the repository config file you created for your
+satellite instance and reference the config map in you BuildConfig's input/source config map list
+- With Satellite, and Docker build strategy, you have to modify your Dockerfile and manually copy and initialize as 
+needed the information from the base image as well as your repository config file for your satellite instance
+- And remember to set the skip layers optimization option (i.e. squash layers) with Docker Strategy builds
+
+This enhancement proposal aims to introduce more centralized and automated infrastructure to ease the burden of 
+using OpenShift Builds with RHEL subscriptions/entitlements.
+
+## Motivation
+
+Simplify overall usability for using RHEL Subscriptions/Entitlements with OpenShift Builds, as we took a hit in 
+4.x vs. 3.x.
+
+### Goals
+
+No longer require (though still allow) manual manipulation of the BuildConfig to consume entitlements.
+
+No longer require (though still allow) manual injection of subscription credentials into a user's namespace.
+
+### Non-Goals
+
+Reach a V3.x level, where no post install activity is needed to consume entitlements.
+
+In other words, no changes to the V4.x installer to allow the user to inject the subscription
+particulars.
+
+## Proposal
+
+### User Stories [optional]
+
+No new user stories are introduced here.  We are improving the user's 4.x experience for existing stories.
```

Comment:
I have customers who are using Pods to build their image, the entrypoint is a script that installs software from entitled sources. They are not using BuildConfigs. Are we considering this use-case as well?  Having an annotation not only for the BuildConfig but for Pods as well?

Reply:
No, that is not a use case that is being considered as part of this enhancement.  There are of course solutions customers can already use for that:

a) use the MCO+CRIO to distribute content to the nodes and then inject it into the pods
b) write an admission control webhook that adds mounts to pods (presuming they have chosen a way to distribute the mountable content into the namespaces)


But the main challenge for builds is that we need to take content that is in the pod (however we decide to get it into the pod) and pass it on to buildah as a mount so that the content is available to the Dockerfile but does not get included in the committed/published output image.

Global distribution+mounting of content universally across pods is, frankly, a use case that's beyond the scope of the team doing this work and trying to solve it satisfactorily would hold up the delivery of this much needed build feature (much as we'd love to have such a thing already exist, since it would make this enhancement much simpler).

Reply:
I would like a story that says:

"As a multi-cluster administrator, I can enable my cluster for entitlements via a declarative API that works well with gitops (or similar patterns) so that enabling entitled builds is not a burden as I scale the number of clusters under management".

Reply:
OK, adding existing stories per @bparees , adding new story per @derekwaynecarr (with the preliminary note that some implementation details for that story are for the time being missing), and a note regarding @zvonkok 's store which has been deemed out of scope by @bparees will be part of next push.